Description: no pie for relocatable LD calls, with toolchain defaulting to pie
Author: Dimitri John Ledkov <xnox@ubuntu.com>

Index: qemu/rules.mak
===================================================================
--- qemu.orig/rules.mak
+++ qemu/rules.mak
@@ -93,7 +93,12 @@ module-common.o: CFLAGS += $(DSO_OBJ_CFL
 	$(if $(findstring /,$@),$(call quiet-command,cp $@ $(subst /,-,$@), "  CP    $(subst /,-,$@)"))
 
 
-LD_REL := $(CC) -nostdlib -Wl,-r $(LD_REL_FLAGS)
+ifeq (,$(shell $(CC) -fsyntax-only -no-pie -xc /dev/null 2>&1))
+no_pie:=-no-pie
+else
+no_pie:=
+endif
+LD_REL := $(CC) $(no_pie) -nostdlib -Wl,-r $(LD_REL_FLAGS)
 
 %.mo:
 	$(call quiet-command,$(LD_REL) -o $@ $^,"  LD -r $(TARGET_DIR)$@")
